frontend/src/components/SearchBar.tsx

```diff
@@ -9,71 +10,74 @@ const SearchBar: React.FC<SearchBarProps> = ({ onSearch }) => {
               </label>
-              <hr className='w-48 h-0.5 bg-[#5F5F5F] place-self-center'></hr>
+              <hr className={`w-48 h-0.5 bg-[${colors.dark_gray}] place-self-center`}></hr>
               <label className="label cursor-pointer flex py-2 px-3 items-center gap-4 pt-3.5">
-                <input type="checkbox" className="checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[#9A9A9A]" />
-                <span className="label-text flex text-[#5F5F5F]">Monday</span>
+                <input type="checkbox" className={`checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[${colors.med_gray}]`} />
+                <span className={`label-text flex text-[${colors.dark_gray}]`}>Monday</span>
               </label>
               <label className="label cursor-pointer flex py-2 px-3 items-center gap-4">
-                <input type="checkbox" className="checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[#9A9A9A]" />
-                <span className="label-text flex text-[#5F5F5F]">Tuesday</span>
+                <input type="checkbox" className={`checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[${colors.med_gray}]`} />
+                <span className={`label-text flex text-[${colors.dark_gray}]`}>Tuesday</span>
               </label>
               <label className="label cursor-pointer flex py-2 px-3 items-center gap-4">
-                <input type="checkbox" className="checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[#9A9A9A]" />
-                <span className="label-text flex text-[#5F5F5F]">Wednesday</span>
+                <input type="checkbox" className={`checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[${colors.med_gray}]`} />
+                <span className={`label-text flex text-[${colors.dark_gray}]`}>Wednesday</span>
               </label>
               <label className="label cursor-pointer flex py-2 px-3 items-center gap-4">
-                <input type="checkbox" className="checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[#9A9A9A]" />
-                <span className="label-text flex text-[#5F5F5F]">Thursday</span>
+                <input type="checkbox" className={`checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[${colors.med_gray}]`} />
+                <span className={`label-text flex text-[${colors.dark_gray}]`}>Thursday</span>
               </label>
               <label className="label cursor-pointer flex py-2 px-3 items-center gap-4">
-                <input type="checkbox" className="checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[#9A9A9A]" />
-                <span className="label-text flex text-[#5F5F5F]">Friday</span>
+                <input type="checkbox" className={`checkbox flex flex-col items-start w-4 h-4 rounded-sm border-2 border-[${colors.med_gray}]`} />
```

Comment:
you can factor out these input and span elements into a component for re usability instead of repeatedly declaring it